icepack/models/mass_transport.py

```diff
@@ -26,19 +26,11 @@
 
 class Continuity:
     r"""Describes the form of the mass continuity equation"""
-    def __init__(self, dimension):
-        if dimension == 2:
-            self.facet_normal = firedrake.FacetNormal
-            self.grad = firedrake.grad
-            self.div = firedrake.div
-            self.ds = firedrake.ds
-        elif dimension == 3:
-            self.facet_normal = utilities.facet_normal_2
-            self.grad = utilities.grad_2
-            self.div = utilities.div_2
-            self.ds = firedrake.ds_v
-        else:
-            raise ValueError('Dimension must be 2 or 3!')
+    def __init__(self):
+        self.facet_normal = utilities.facet_normal_nd
```

Comment:
Since this object doesn't get the dimension in its initializer anymore, there isn't much reason to set these member variables when you could just call `grad_nd` and so forth directly in the `__call__` method. At that rate, there isn't much reason why this should even be a class; it could just be a function. The only change below would be a conditional to use either `ds` or `ds_v` depending on whether or not the mesh was extruded.

Reply:
Leaving as a class for now.